apps/web/src/components/Button/ButtonWithLink.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import Link, { LinkProps } from 'apps/web/node_modules/next/link';
+import { Button, ButtonProps } from './Button';
+
+export type ButtonWithLinkProps = Omit<ButtonProps, 'href' | 'className'> & {
+  href: string;
+  linkClassNames?: string;
+  linkProps?: Omit<LinkProps, 'href' | 'passHref'>;
+  target?: string;
+  rel?: string;
+  buttonClassNames?: string;
+};
+
+export function ButtonWithLink({
+  href,
+  target,
+  rel,
+  linkClassNames,
+  linkProps,
+  variant,
+  size,
+  children,
+  buttonClassNames,
+  rounded = false,
+  fullWidth = false,
+  disabled = false,
+  isLoading = false,
+  ...buttonProps
+}: ButtonWithLinkProps) {
+  return (
+    <Link
+      href={href}
+      target={target}
+      rel={rel}
+      className={`${linkClassNames}${disabled ? ' pointer-events-none' : ''}`}
```

Comment:
nitpick: could use classNames with conditional rendering here

Reply:
also ideal world we'd just apply the styles from Button onto the Link instead of wrapping a button. but IMO works for now!